### DISCUSSION TITLE
ROS2 and Webots-Crazyflie2 environment connection

### DISCUSSION THREAD

**Discussion by ntukenmez3** [bitcraze/discussions#887] [Q&A] [answered]
We would like to implement our controller developed by our team on CF2 simulator. However, we could not connect ros2 with webots cf2 simulator. Other simulator environments like Mavic can be connected but we could not the same process for CF2. Thank you for your help.

**Comment by knmcguire** [comment, marked as answer]
Just summarizing what I've mentioned to you on linkedin: There is no official ROS2 webots node for the Crazyflie. Its in the pipeline but it is very uncertain when this will be released. It's better if you are in  a hurry to implement it yourself.

This is a good tutorial you can look at from cyberbotics, where you can replace nodes with the crazyflie model: First follow this tutorial and than just start replacing things with the crazyflie https://docs.ros.org/en/rolling/Tutorials/Advanced/Simulators/Webots/Setting-Up-Simulation-Webots-Basic.html 

If that doesn't work out you can look for inspiration in my personal repo here where I've connected the crazyflie webots simulator to ros2. Beware that it might already be outdated already: https://github.com/knmcguire/crazyflie_ros2_experimental

If you have any more questions please ask them here on this platform, as this is the right place for support.

**Comment by ntukenmez3** [comment]
I’m so thankful to you, I've already started to use new packages on ros2. If we can get promising results with real cf2, we'll let you know, thank you so much again, have a great day.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
